pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go

```
@@ -192,10 +191,9 @@ func (pl *ServiceAffinity) RemovePod(ctx context.Context, cycleState *framework.
 func getPreFilterState(cycleState *framework.CycleState) (*preFilterState, error) {
 	c, err := cycleState.Read(preFilterStateKey)
 	if err != nil {
-		// The metadata wasn't pre-computed in prefilter. We ignore the error for now since
-		// Filter is able to handle that by computing it again.
-		klog.V(5).Infof(fmt.Sprintf("reading %q from cycleState: %v", preFilterStateKey, err))
-		return nil, nil
+		// The metadata wasn't pre-computed in prefilter.
+		klog.V(5).Infof(fmt.Sprintf("Error reading %q from cycleState: %v", preFilterStateKey, err))
+		return nil, err
```

Ditto. We can simply return a verbose err and remove the logging.